src/common.js

```diff
@@ -17,7 +17,10 @@ export const checkStatus = (response) => {
 
   let error = new Error(response.statusText);
   error.response = response;
-  throw error;
+
+  return response.json()
+    .then(json => Promise.reject(Object.assign(error, json)))
```

Comment:
Actually, now I see what's happening: it *does* trigger the `.catch` but it still works because `Object.assign` modifies its first argument (`error`) directly, so `Promise.reject(error)` will still contain the correct data when it gets there. 
Seems fine, except since we don't use the `e => ...` argument in the subsequent catch we might as well just replace that with `() => ...`.